core/src/main/java/io/grpc/transport/MessageFramer.java

```diff
@@ -237,7 +237,7 @@ public void dispose() {
 
   private void commitToSink(boolean endOfStream, boolean flush) {
     if (buffer == null) {
-      buffer = bufferAllocator.allocate(0);
+      buffer = bufferAllocator.empty();
```

Comment:
Maybe that doesn't work since the `allocate` method is only given a hint ... nevermind :)